tests/test_stochatreat_input_output.py

```diff
@@ -217,3 +219,21 @@ def test_stochatreat_output_no_null_treats(treatments_dict):
     treatments_df = treatments_dict["treatments"]
     asrtmsg = "There are null assignments"
     assert treatments_df["treat"].isnull().sum() == 0, asrtmsg
+
+
+@pytest.mark.parametrize("misfit_strategy", ["global", "stratum"])
+def test_stochatreat_output_index_content_unchanged(treatments_dict, misfit_strategy):
+    """Tests that the functions's output's index column matches the input index column"""
+    data_with_random_index = treatments_dict["data"]
+    data_with_random_index = data_with_random_index.set_index(
+        pd.Index(np.random.choice(300, 100, replace=False))
+    )
+    treatments = stochatreat(
+        data=data_with_random_index,
+        block_cols= treatments_dict["block_cols"],
+        treats=treatments_dict["n_treatments"],
+        idx_col=treatments_dict["idx_col"],
+        misfit_strategy=misfit_strategy,
+    )
+    asrtmsg = "The output and input indices do not have the same content"
+    assert set(treatments.index) == set(data_with_random_index.index), asrtmsg
```

Comment:
That's a pretty weak test. Could make that much stronger, I think:

1. construct a DataFrame with random but unique index
2. Also assign a unique but random index column
3. pass that to stochatreat in various configs
4. check in the end that the index values are the same
5. check that the correspondence between index and index column are unperturbed